drivers/pinmux/CMakeLists.txt

```diff
@@ -1,6 +1,11 @@
 # SPDX-License-Identifier: Apache-2.0
 
-zephyr_library()
```

Comment:
this CMakeLists is only included if `CONFIG_PINMUX=y`, so if this library gets no sources, it means that there must be an issue somewhere else.

Reply:
yes, and that issue is because several boards are defining `CONFIG_PINMUX=y`, even when not selecting a driver from `drivers/pinmux`.

Maybe because of this guide:
https://docs.zephyrproject.org/latest/guides/porting/board_porting.html?highlight=config_pinmux#create-your-board-directory

> CMakeLists.txt: if you need to add additional source files to your build.
>
> One common use for this file is to add a pinmux.c file in your board directory to the build, which configures pin > controllers at boot time. In that case, CMakeLists.txt usually looks like this:
>
> ```
> if(CONFIG_PINMUX)
>  zephyr_library()
>  zephyr_library_sources(pinmux.c)
> endif()
> ```

so following that guide will in most cases result in the mess this PR is trying to clean up.

Now, one way of fixing that would be to avoid having `CONFIG_PINMUX=y` just because you add `pinmux.c`, and that would be identical to option 4 here:
https://github.com/zephyrproject-rtos/zephyr/pull/38608#issuecomment-922905032